src/gridcoin/researcher.cpp

```diff
@@ -1195,6 +1195,14 @@ void Researcher::Reload(MiningProjectMap projects, GRC::BeaconError beacon_error
 
     if (mining_id.Which() != MiningId::Kind::CPID) {
         for (const auto& project_pair : projects) {
+            // Stop searching if the current mining_id already has an active beacon
+            if (const CpidOption cpid = mining_id.TryCpid()) {
+                LOCK(cs_main);
```

Comment:
I don't think you need a lock on cs_main here. cs_main is always supposed to be taken before calling any of these functions, and putting in an extra lock might seem harmless, but in fact could cause a deadlock if another lock from another thread gets taken in between.

Reply:
Right, I see that's documented in the .h file. I have removed the lock now.